Common/Securities/NullBuyingPowerModel.cs

```diff
@@ -0,0 +1,33 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+namespace QuantConnect.Securities
+{
+    /// <summary>
+    /// Provides a buying power model considers that there is sufficient buying power for all orders
+    /// </summary>
+    public class NullBuyingPowerModel : BuyingPowerModel
```

Comment:
Would of expected it to use the https://github.com/QuantConnect/Lean/blob/master/Common/Securities/ConstantBuyingPowerModel.cs with a small value because it's not only about placing a new trading but the margin used to maintain it without getting a margin call?

Reply:
Users could use the `MarginCallModel.Null` to handle the margin calls. Regardless, I  added the `NullBuyingPowerModel.GetMaintenanceMargin` returning a `MaintenanceMargin` of zero. It means the total margin used is always zero and margin calls will not be triggered.

This feature is inspired by the  `ConstantBuyingPowerModel`. I didn't inherit from that class, because `ConstantBuyingPowerModel.GetInitialMarginRequirement` gives us a very small initial margin that leads to a very large quantity if we use `SetHoldings` or `CalculateOrderQuantity`. If the algorithm is using `SetHoldings` the author would have to rewrite it.